content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,586 @@
+    interface CustomEvents {
+      Name: NameEvent;  // Matches `on:Name`
+    }
+    interface CustomCaptureEvents {
+      Name: NameEvent;  // Matches `oncapture:Name`
+    }
+  }
+}
+
+// Usage
+<div on:Name={(event) => console.log("name is", event.detail.name)} />;
+```
+
+<Aside>
+**Note**: By default, using native events like `mousemove` with the `on` prefix (e.g., <div on:mousemove={e => {}} />) will result in a TypeScript error. 
+This is because they are not included in the custom event typings for Solid.
+To solve this, extend the `CustomEvents` and `CustomCaptureEvents` interfaces to include events from the `HTMLElementEventMap`:
+
+To include all native events:
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends HTMLElementEventMap {}
+    interface CustomCaptureEvents extends HTMLElementEventMap {}
+  }
+}
+```
+
+To include specific native events, you can pick certain events (e.g. `mousemove` and `pointermove`):
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends Pick<HTMLElementEventMap, "mousemove" | "pointermove"> {}
+  }
+}
+```
+</Aside>
+
+#### Forcing properties and custom attributes 
+
+Solid's [`prop:___`]() allows forcing properties, while [`attr:___`]() permits custom attributes. 
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface ExplicitProperties {
+      count: number;
+      name: string;
+    }
+    interface ExplicitAttributes {
+      count: number;
+      name: string;
+    }
+  }
+}
+
+// Usage
+<Input prop:name={name()} prop:count={count()}/>
+<my-web-component attr:name={name()} attr:count={count()}/>
+```
+
+#### Custom directives
+
+Custom directives can be applied using the [`use:___`]() attribute.
+
+```typescript
+function model(element: HTMLInputElement, value: Accessor<Signal<string>>) {
+  const [field, setField] = value();
+  createRenderEffect(() => (element.value = field()));
+  element.addEventListener("input", (e) => {
+    const value = (e.target as HTMLInputElement).value;
+    setField(value);
+  });
+}
+
+declare module "solid-js" {
+  namespace JSX {
+    interface Directives {
+      model: Signal<string>;  // Corresponds to `use:model`
+    }
+  }
+}
+
+// Usage
+let [name, setName] = createSignal("");
+<input type="text" use:model={[name, setName]} />;
+```
+
+##### Addressing import issues with directives
+
+When using directives imported from another file/module, TypeScript may eliminate the import if it mistakenly perceives it as a type. 
+
+* Set the `onlyRemoveTypeImports: true` option in `babel-preset-typescript`. 
+If using `vite-plugin-solid`, this can be configured with `solidPlugin({ typescript: { onlyRemoveTypeImports: true } })` in `vite.config.ts`. 
+However, this approach mandates judicious use of `export type` and `import type`.
+* Including a statement like `false && d;` in the importing module ensures TypeScript retains the directive's import. 
```

Comment:
I'm not sure if anyone will understand what `false && d;` is supposed to mean.
A simple example like this might be clearer:
```tsx
import { directive } from "./directives"

directive // prevents TypeScript's tree-shaking

<div use:directive />
```